plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiBackgroundSplitLoader.java

```diff
@@ -74,7 +74,7 @@ public void start()
                             .map(partition -> Futures.submit(() -> loadSplits(partition), executor))
                             .peek(this::hookErrorListener)
                             .collect(Collectors.toList());
-                    Futures.whenAllComplete(futures).run(asyncQueue::finish, directExecutor());
+                    hookErrorListener(Futures.whenAllComplete(futures).run(asyncQueue::finish, directExecutor()));
```

Comment:
@wendigo Shouldn't the hook on line 81 be sufficient?

Reply:
It’s about ignoring return value of the combiner call. What would you propose instead adding a hook?

Reply:
Got it, thanks

Reply:
I’ve migrated this change to other PR. Seems to work just fine